File: binarySearch/searchInsertPosition.py
# https://leetcode.com/problems/search-insert-position

class Solution:
    def searchInsert(self, nums: List[int], target: int, lo=0, hi=None) -> int:
        if lo < 0:
            raise ValueError('lo must be non-negative')
        if hi is None:
            hi = len(nums)
        while lo < hi:
            mid = (lo+hi)//2
            if nums[mid] < target: lo = mid+1
            else: hi = mid
        return lo

File: binarySearch/searchInsertPosition.js
// https://leetcode.com/problems/search-insert-position

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
const searchInsert = (nums, target, lo = 0, hi = null) => {
    if (lo < 0) {
        throw new Error('lo must be non-negative');
    }
    if (hi === null) {
        hi = nums.length;
    }
    while (lo < hi) {
        let mid = Math.floor((lo + hi)/2);
        if (nums[mid] < target) {
            lo = mid + 1
        } else {
            hi = mid
        }
    }
    return lo;
};
